[nickt121] iP

nickt121 opened the pull request:
### ProcrastinAid

> I dont wanna do work anymore -me

ProcrastinAid helps you **stop** procrastinating, or maybe it helps you procrastinate instead? Who knows?

- Text based (for now)
- Easy to use

All you need to do is,

1. download it from [here](https://github.com/nickt121/ip/releases/download/Level-9/procrastinaid.jar).
2. run it with `java -jar "procrastinaid.jar"`
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:
- [x] Managing tasks
- [x]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

nickt121 committed:
feat: refactored Duke to ProcrastinAid

nickt121 committed:
feat: refactor greeting and bye into individual methods

nickt121 committed:
feat: echo user input

nickt121 committed:
feat: refactor userinput into method and added formating

nickt121 committed:
feat: added feature to add to and view tasks in a list

nickt121 committed:
feat: new class for individual task

nickt121 committed:
feat: format responses when adding and removing from list

nickt121 committed:
feat: use new Task class in main loop

nickt121 committed:
feat: mark and unmark task as complete

nickt121 committed:
refactor: changed Task to override toString instead of using getter

nickt121 committed:
feat: change task to abstract and add new child classes

nickt121 committed:
fix: string formating

nickt121 committed:
feat: support functions for new classes

nickt121 committed:
refactor: cleanup code

nickt121 committed:
test: add text ui test cases

nickt121 committed:
feat: add procrastinaidexception error class

nickt121 committed:
feat: error handling and raising custom error

nickt121 committed:
feat: delete task feature

nickt121 committed:
feat: add and use tasktype enum

nickt121 committed:
feat: add writing and reading of tasks to file

Add new feature by adding new Storage class.

New feature achieved through the use of new storage class and taskfactory factory class

nickt121 committed:
feat: use new storage class in main app

nickt121 committed:
Merge branch 'branch-Level-7'

nickt121 committed:
fix: write new line to csv file after each task

nickt121 committed:
feat: parse input datetime as LocalDateTime

nickt121 committed:
Merge branch 'branch-Level-8'

nickt121 committed:
misc: add task.txt to gitignore

nickt121 committed:
feat: refactor into parser, storage, Ui class

nickt121 committed:
misc: untrack tasks.txt from git

Ben926 commented on src/main/java/ProcrastinAid.java:
```diff
@@ -0,0 +1,97 @@
+
+public class ProcrastinAid {
+    private static Storage storageFile = new Storage("./tasks.txt");
+    private static TaskList taskList = storageFile.loadFromFile();
```
Perhaps you can rename this as tasks to prevent confusion between the class and object name?

Ben926 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,55 @@
+import java.util.Scanner;
+
+public class Storage {
+    private final String DATAFILE;
```
Perhaps the variable name should specify that this is a path to the txt file, as this sounds like it would be a instance of the File class.

Ben926 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,30 @@
+public abstract class Task {
+    private final String description;
+    private boolean isDone;
```
I like that you phrased boolean variable names in your code appropriately!

Ben926 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    public Task markTaskAsDone(int index, boolean isDone) {
+        Task temp = this.tasks.get(index);
```
Perhaps you can rename your temp variables as tempTask?

Ben926 reviewed (commented):
Overall, very good job with the code. Very readable mostly and followed good coding standards.

nickt121 committed:
feat: refactor all class into packages

nickt121 committed:
feat: refactor readme files to point to new paths

nickt121 committed:
Merge branch 'add-gradle-support'

nickt121 committed:
feat: updated build.gradle to build project

nickt121 committed:
feat: junit test for parser, event and todo class

nickt121 committed:
feat: add java docs

nickt121 committed:
Merge branch 'branch-A-JavaDoc'

nickt121 committed:
fix: edit to comply with coding standards

nickt121 committed:
Merge branch 'branch-A-CodingStandard'

nickt121 committed:
feat: add findtask feature

Add a new feature find to the application and a new method findTasks to tasklist to implement the new feature

nickt121 committed:
Merge branch 'branch-Level-9'

nickt121 committed:
fix: rename variable to fix refactoring issue

yuqing-tham commented on src/main/java/procrastinaid/task/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+
+import procrastinaid.exception.ProcrastinAidException;
+
+public class Deadline extends Task {
```
Would appreciate it if there are header comments for all public classes and methods!

yuqing-tham commented on src/main/java/procrastinaid/task/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+
+    private LocalDateTime by;
+
+    public Deadline(String description, boolean isDone, String by) throws ProcrastinAidException {
```
Good job on the naming for boolean variable `isDone`!

yuqing-tham commented on src/main/java/procrastinaid/task/Event.java:
```diff
@@ -0,0 +1,43 @@
+package procrastinaid.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import procrastinaid.exception.ProcrastinAidException;
```
I like how the ordering of the imports is very consistent in all the classes.

yuqing-tham commented on src/main/java/procrastinaid/task/Storage.java:
```diff
@@ -0,0 +1,71 @@
+ * Storage class to handle saving and loading of tasks to and from a file.
+ */
+public class Storage {
+    private final String FILEPATH;
```
Good job on the all caps for constants.

yuqing-tham commented on src/main/java/procrastinaid/task/Task.java:
```diff
@@ -0,0 +1,54 @@
+
+    /**
+     * Returns the status icon of the procrastinaid.task isDone attribute. The icon is either a cross or a blank.
+     *
```
Kudos for not making a common mistake of forgetting the empty line between the description and the parameter in the header comment.

yuqing-tham commented on src/main/java/procrastinaid/task/TaskFactory.java:
```diff
@@ -0,0 +1,34 @@
+     * @throws ProcrastinAidException If the file line is invalid.
+     */
+    public static Task createTaskFromFile(String fileLine) throws ProcrastinAidException {
+        String[] parts = fileLine.split(",");
```
Good job on using plural form for an array.

yuqing-tham commented on src/main/java/procrastinaid/ui/ProcrastinAid.java:
```diff
@@ -0,0 +1,106 @@
+            String command = parser.getCommand();
+            String arguments = parser.getRawArgs();
+            try {
+                switch (command) {
+                case "bye":
```
Good catch on having case with same amount of spacing as switch, as per the course's coding standard.

yuqing-tham commented on src/main/java/procrastinaid/ui/ProcrastinAid.java:
```diff
@@ -0,0 +1,106 @@
+        Ui.showTaskListSize(tasks.getSize());
+    }
+
+    public static void markTaskAsDone(String taskNumber, boolean done) {
```
Perhaps `done` can be renamed to follow the coding standard?

yuqing-tham reviewed (commented):
LGTM. Just some minor coding standard violations could be improved on. Great job!

nickt121 committed:
misc: add checkstyle config xml files

nickt121 committed:
Merge branch 'branch-CheckStyle'

nickt121 committed:
fix: fix codestyle violations

nickt121 committed:
misc: move checkstyle files to config folder

nickt121 committed:
feat: implement basic gui without logic

nickt121 committed:
feat: implement logic for gui

nickt121 committed:
Merge branch 'branch-Level-10'

nickt121 committed:
feat: add assertions

Add assertions to Storage, setCUrrentInput, getResponse and addTask

This methods all rely on crucial inputs that should not be null under any circumstances.

Adding assertions prevent that.

nickt121 committed:
misc: improve code quality

Fix code quality issues in Tasks classes and main app class

There are several issues with the code quality of these classes. Removed magic numbers as well as improve abstraction to comply with SLAP

nickt121 committed:
Merge pull request #2 from nickt121/branch-A-Assertions

feat: add assertions

nickt121 committed:
Merge branch 'master' into branch-A-CodeQuality

nickt121 committed:
Merge pull request #3 from nickt121/branch-A-CodeQuality

misc: improve code quality

nickt121 committed:
misc: setup ci/cd

nickt121 committed:
fix: fix code style issues

nickt121 committed:
feat: add tag attribute to all tasks classes

nickt121 committed:
feat: allow searching by tag in find command

nickt121 committed:
feat: increase robustness of commands

nickt121 committed:
fix: fix junit test cases

nickt121 committed:
feat: add icon and title bar

nickt121 committed:
feat: change user and bot icons

nickt121 committed:
feat: add user guide and ui picture

nickt121 committed:
feat: improve user guide

nickt121 committed:
fix: handle invalid task number format

nickt121 committed:
fix: handle missing tag in tag command

nickt121 committed:
feat: anchor gui elements
